mailstream/mailstream.php

```diff
@@ -114,16 +105,40 @@ function mailstream_generate_id($uri)
 	return $message_id;
 }
 
+function mailstream_send_hook(&$a, $data)
+{
+	$criteria = array('uid' => $data['uid'], 'contact-id' => $data['contact-id'], 'uri' => $data['uri']);
+	$item = Post::selectFirst([], $criteria);
+	if (empty($item)) {
+		Logger::error('mailstream_send_hook could not find item');
+		return;
+	}
+
+	$user = User::getById($item['uid']);
+	if (empty($user)) {
+			Logger::error('mailstream_send_hook could not fund user', ['uid' => $item['uid']]);
+		return;
+	}
+
+	if (!mailstream_send($data['message_id'], $item, $user)) {
+		$delayed = date(DateTimeFormat::utc('now + 1 hour'));
+		$data['tries'] += 1;
+		Hook::fork(['priority' => PRIORITY_LOW, 'delayed' => $delayed], 'mailstream_send_hook', $data);
```

Comment:
Please have a look here: https://github.com/friendica/friendica/blob/2021.06-rc/src/Worker/APDelivery.php#L104

I guess that `Worker::defer()` can help here. it automatically calculates the time between retrials, so you don't need to have a look at that as well.